src/backend/mpi/swdsm.cpp

```diff
@@ -769,17 +769,20 @@ void * loadcacheline(void * x){
 
 		unsigned long offset = getOffset(lineAddr);
 		if(isPowerOf2((tempsharer)&invid) && tempsharer != id && prevsharer == 0){ //Other private. but may not have loaded page yet.
-			long ownid = (long)(tempsharer&invid);
-			unsigned long owner;
-			for(n=0; n<numtasks; n++){
-				if((unsigned)(1<<n)==(unsigned)ownid){
+			unsigned long ownid = tempsharer&invid; // remove own bit
+			static const unsigned long invalid_owner = static_cast<unsigned long>(-1);
```

Comment:
Can this be in some larger scope to avoid having to create it twice? Also, the static is fairly useless, if not harmful, since any semi-decent compiler will constant propagate the value.

I would suggest using a constexpr in global scope: `constexpr unsigned long invalid_owner = static_cast<...>(-1);`

I assume that the static cast is used to silence compiler warnings.

Reply:
resolved in 03424b5 and 3e1353d